packages/hash/integration/src/tests/model/knowledge/link.model.test.ts

```diff
@@ -150,4 +169,183 @@ describe("Link model class", () => {
 
     expect(links).toHaveLength(0);
   });
+
+  let playlistEntityType: EntityTypeModel;
+
+  let playlistEntity: EntityModel;
+
+  let songEntityType: EntityTypeModel;
+
+  let songEntity1: EntityModel;
+
+  let songEntity2: EntityModel;
+
+  let songEntity3: EntityModel;
+
+  let hasSongLinkType: LinkTypeModel;
+
+  let hasSongLink1: LinkModel;
+
+  let hasSongLink2: LinkModel;
+
+  let hasSongLink3: LinkModel;
+
+  it("can create an ordered link", async () => {
+    playlistEntityType = await createEntityType({ title: "Playlist" });
```

Comment:
It's a bit strange to me that you're storing the `EntityTypeModel`s outside of the test case, but initializing them in `it` instead of in the `beforeAll` or start of the describe like is done elsewhere. This is inconsistent and possibly wrong?

Reply:
The reason I initialised them here is to separate the test entities/links for ordered links from the prior ones which are unordered. 

Also to be clear are you only referring to the types that are created here? I don't think it's a good idea to initialise the links in the `beforeAll` as we would then be calling `LinkModel.create` before the "can create link" test. If an error is thrown in that method, it won't be the test that catches it - it'll be the `beforeAll`. I think this could make it harder to debug.

Reply:
> but initializing them in it instead of in the beforeAll or start of the describe like is done elsewhere. This is inconsistent and possibly wrong?

There are some things we initialise within tests in other test files as well, so I didn't see it as inconsistent. If you see it as wrong could you explain why? In which case we may need to revise all the test files in that case and address it in a separate PR.

Reply:
> The reason I initialised them here is to separate the test entities/links for ordered links from the prior ones which are unordered.

That makes sense, but at that point I think we should probably be splitting up into two `describe`s. As otherwise the grouping of the test cases is unclear.

> Also to be clear are you only referring to the types that are created here

Yes, I'm basically referring to the context the test is ran in, or the initial starting state if you'd like. In a lot of the other files we set up the initial starting state in the `beforeAll`, but I now see that we mix and match this behaviour, not just in this PR but in some of the other files.

> In which case we may need to revise all the test files in that case and address it in a separate PR.

I think this might be what we have to do. We've got a variety of inconsistent patterns, with 
* Setting up initial state in the `beforeAll` and storing the types globally for the file. 
* We've got places where the types are declared in the describe and then initialised at the start of the describe before the test cases. 
* We've got places where the types are declared in the describe and then initialised _across_ the test cases. 
* We've got places where types are declared _in between_ test cases and declared before _or_ in them.

I'm not suggesting we need to pick only **one**,  but it's definitely inconsistent and a bit confusing as it stands now.

I'd like to see us reach a pattern where it is clear what conditions and starting state is needed for a test, and what the test is clearly doing. But to do that, I think we'll need transactions or something else to ensure a clean DB sandbox for each set of tests.

Reply:
As an action on this PR, perhaps we do nothing in the code, and open an Asana task to capture the above

Reply:
[Asana task here](https://app.asana.com/0/1202805690238892/1203055997033804/f), leaving the thread open in case GitHub decides to be a pain